PVTPackage/source/MultiphaseSystem/PhaseModel/BlackOil/BlackOil_WaterModel.cpp

```diff
@@ -26,13 +26,18 @@ BlackOil_WaterModel::BlackOil_WaterModel( const std::vector< double > & PVTW,
   m_surfaceMassDensity( waterSurfaceMassDensity ),
   m_surfaceMolecularWeight( waterSurfaceMolecularWeight )
 {
-  m_PVTW.ReferencePressure = PVTW[0];
-  m_PVTW.Bw = PVTW[1];
-  m_PVTW.Compressibility = PVTW[2];
-  m_PVTW.Viscosity = PVTW[3];
-
-  //Density
-  m_surfaceMoleDensity = m_surfaceMassDensity / m_surfaceMolecularWeight;
+  // if water is present, PVTW.size() == 4
+  // if water is absent (for two-phase dead-oil), PVTW.size() == 0 => we skip initialization
```

Comment:
> we skip initialization

And is it ok to live with a non initialized `m_PVTW ` or should we die badly?

Reply:
I modified this constructor. Now it is only used when water is present in the system, so just check the size of `PVTW` and I error out if it is not 4.